### ISSUE TITLE
Add RandomFrequenceMask transform

### ISSUE THREAD

**Issue by guarin** [lightly-ai/lightly#1660]
See https://github.com/lightly-ai/lightly/issues/1646 for information

**Comment by payo101** [comment]
Hi @guarin , I'm pretty new to open source and would like to contribute to this. I have been looking at the Contributions.md for basic guidelines, where should I go next on this?
Also are you on the hacktober's discord? (If it's comfortable with you to disclose that? If I had some doubts, I thought I could contact you their itself)

**Comment by philippmwirth** [comment]
Hi @payo101, the next steps are: 1) We'll assign the issue to you; 2) You open a pull request as outlined in [CONTRIBUTIONS.md](https://github.com/lightly-ai/lightly/blob/master/CONTRIBUTING.md#pull-requests). For more information on the implementation details please check out the issue referenced in the description. For reference you can also take a look at the augmentations we have already implemented (see, e.g. [random rotation](https://github.com/lightly-ai/lightly/blob/master/lightly/transforms/rotation.py)). Could you please quickly confirm if you're interested in working on this?

> Also are you on the hacktober's discord? (If it's comfortable with you to disclose that? If I had some doubts, I thought I could contact you their itself)

Forwarding this to @guarin :)

**Comment by payo101** [comment]
Yeah I am

**Comment by guarin** [comment]
> Also are you on the hacktober's discord? (If it's comfortable with you to disclose that? If I had some doubts, I thought I could contact you their itself)

Yes you can find me under the same name :)

We also have a discord for lightly related questions: https://discord.gg/xvNJW94

**Comment by payo101** [comment]
Hi @guarin , So I have 2 doubts to ask
- Will the input be the RFFT of the Image, or the Image itself? I have been currently working with assuming the former
- Should the output also return the binary mask that's being applied on the FFT?

**Comment by payo101** [comment]
Also, I am facing a mypy error while proceeding to commit
<img width="835" alt="Screenshot 2024-10-06 235119" src="https://github.com/user-attachments/assets/5dd0a599-01ce-4c40-998e-8dd68469b1b3">

After looking into it, all necessary steps that can be done to prevent this has been done, but it's still showing this error, any chance anyone could help me with it?

**Comment by philippmwirth** [comment]
> Also, I am facing a mypy error while proceeding to commit <img alt="Screenshot 2024-10-06 235119" width="835" src="https://private-user-images.githubusercontent.com/35198092/373970238-5dd0a599-01ce-4c40-998e-8dd68469b1b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgyODI1NTcsIm5iZiI6MTcyODI4MjI1NywicGF0aCI6Ii8zNTE5ODA5Mi8zNzM5NzAyMzgtNWRkMGE1OTktMDFjZS00YzQwLTk5OGUtOGRkNjg0NjliMWIzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA3VDA2MjQxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjNzA0ZWI0YjRiNjNmOWRjZWYwNzgxMWRkMGNlNGMyNzhhN2FlNWViMDEwZDRlZmZhYjQ5NTEzY2QxY2NiZjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QetWKPaZAKhrjxkhejqLUbLMuQiQUDP3K2NTmnFOg48">
> 
> After looking into it, all necessary steps that can be done to prevent this has been done, but it's still showing this error, any chance anyone could help me with it?

Hey @payo101, it's interesting that this only comes up now. You can fix it by adding an `ignore` statement:
```python
class RandomResizedCropAndFlip(nn.Module): # type: ignore[misc] # Class cannot subclass "RandomVerticalFlip" (has type "Any")
```
If you look at the other examples just above, this is exactly how we fixed it previously. Strange that `mypy` checks on `master` are passing... Maybe it's also related to `mypy` version. Can you share which version you're using?

**Comment by payo101** [comment]
it's 1.4.1

**Comment by philippmwirth** [comment]
> it's 1.4.1

Ok, that's correct. Did you manage to fix the problem with the proposed `# type: ignore ...`?

**Comment by payo101** [comment]
Yeah, it got fixed  😄

**Comment by payo101** [comment]
> Hi @guarin , So I have 2 doubts to ask
> 
> * Will the input be the RFFT of the Image, or the Image itself? I have been currently working with assuming the former
> * Should the output also return the binary mask that's being applied on the FFT?
Could you also clarify on these two doubts also @philippmwirth

**Comment by philippmwirth** [comment]
You can assume the input is the `RFFT` tensor. No need to return the binary mask.

**Comment by guarin** [comment]
Completed in #1680
